operator/k8s_node.go

```diff
@@ -32,6 +32,30 @@ import (
 )
 
 func runNodeWatcher(nodeManager *allocator.NodeEventHandler) error {
+
+	resourceEventHandler := cache.ResourceEventHandlerFuncs{}
+
+	k8sNodeStore, nodeController := informer.NewInformer(
+		cache.NewListWatchFromClient(k8s.CiliumClient().CiliumV2().RESTClient(),
+			cilium_v2.CNPluralName, v1.NamespaceAll, fields.Everything()),
+		&cilium_v2.CiliumNode{},
+		0,
+		resourceEventHandler,
+		nil,
+	)
+
+	go nodeController.Run(wait.NeverStop)
+
+	if operatorOption.Config.CNPNodeStatusGCInterval != 0 {
+		go runCNPNodeStatusGC("cnp-node-gc", false, k8sNodeStore)
+		go runCNPNodeStatusGC("ccnp-node-gc", true, k8sNodeStore)
+	}
+
```

Comment:
We are already watching the CiliumNode from the execution of https://github.com/cilium/cilium/blob/9061f3d46a7571ca06e113d4ac544f655564c1f9/operator/main.go#L400 so this function can be simplified to this:

```suggestion
	if operatorOption.Config.CNPNodeStatusGCInterval != 0 {
		go runCNPNodeStatusGC("cnp-node-gc", false, ciliumNodeStore)
		go runCNPNodeStatusGC("ccnp-node-gc", true, ciliumNodeStore)
	}
```